source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -447,10 +452,135 @@ bool areAllResponsesSame(const std::vector<Common::Redis::RespValuePtr>& respons
 }
 
 void mgmtNoKeyRequest::onallChildRespAgrregate(Common::Redis::RespValuePtr&& value, int32_t reqindex, int32_t shardindex) {
-  
-   ENVOY_LOG(debug,"response recived for reqindex: '{}', shard Index: '{}'", reqindex,shardindex);
-   ENVOY_LOG(debug, "response: {}", value->toString()); 
-  return;
+  pending_requests_[reqindex].handle_ = nullptr;
+  ENVOY_LOG(debug,"response recived for reqindex: '{}', shard Index: '{}'", reqindex,shardindex);
+  if (reqindex >= static_cast<int32_t>(pending_responses_.size())) {
+        // Resize the vector to accommodate the new index
+        pending_responses_.resize(reqindex + 1);
+  }
+
+  if (value->type() == Common::Redis::RespType::Error){
+    error_count_++;
+    response_index_=reqindex;
+  }
+  // Move the value into the pending_responses at the specified index
+  pending_responses_[reqindex] = std::move(value);
+
+  ASSERT(num_pending_responses_ > 0);
+  if (--num_pending_responses_ == 0) {
+    if (error_count_ > 0 ){
+      if (!pending_responses_.empty()) {
+        Common::Redis::RespValuePtr response = std::move(pending_responses_[response_index_]);
+        callbacks_.onResponse(std::move(response));
+        pending_responses_.clear();
+      }
+    } else {
+      updateStats(error_count_ == 0);
+      if (!pending_responses_.empty()) {
+        if ( pending_requests_[reqindex].rediscommand_ == "pubsub" || pending_requests_[reqindex].rediscommand_ == "keys" || pending_requests_[reqindex].rediscommand_ == "slowlog") {
+              if ((pending_requests_[reqindex].redisarg_ == "numpat" || pending_requests_[reqindex].redisarg_ == "len") && (pending_requests_[reqindex].rediscommand_ == "pubsub" || pending_requests_[reqindex].rediscommand_ == "slowlog")) {
+                  int sum = 0; 
+                  Common::Redis::RespValuePtr response = std::make_unique<Common::Redis::RespValue>();
+                  response->type(Common::Redis::RespType::Integer);
+                  for (auto& resp : pending_responses_) {
+                    if (resp->type() == Common::Redis::RespType::Integer) {
+                      try {
+                          int integerValue = resp->asInteger(); // Convert to integer
+                          if (integerValue >= 0) {
+                              sum += integerValue; // Add integer value to sum if it's non-negative
+                          } else {
+                              // Handle negative integer value
+                              std::cerr << "Error: Negative integer value: " << integerValue << std::endl;
+                              ENVOY_LOG(error, "Error: Negative integer value: {}", integerValue);
+                          }
+                      } catch (const std::exception& e) {
+                          // Handle conversion error
+                          std::cerr << "Error converting integer: " << e.what() << std::endl;
```

Comment:
Error Response has to be sent if there is a unexpected response

Reply:
fixed